backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/DatabaseProcedureService.java

```diff
@@ -33,11 +33,25 @@ public DatabaseProcedureService(
   @Override
   @Transactional(transactionManager = "jpaTransactionManager")
   public Page<Procedure> search(
-      Optional<String> query, DocumentationOffice documentationOffice, Pageable pageable) {
+      Optional<String> query,
+      DocumentationOffice documentationOffice,
+      Pageable pageable,
+      Optional<Boolean> withDocUnits) {
 
     DocumentationOfficeDTO documentationOfficeDTO =
         documentationOfficeRepository.findByAbbreviation(documentationOffice.abbreviation());
 
+    if (withDocUnits.isPresent() && Boolean.TRUE.equals(withDocUnits.get())) {
+      return query
+          .map(
+              queryString ->
+                  repository.findLatestUsedProceduresByLabelAndDocumentationOffice(
+                      queryString.trim(), documentationOfficeDTO, pageable))
+          .orElse(
```

Comment:
Take a look at this: https://www.baeldung.com/java-optional-or-else-vs-or-else-get